sdk/storage/storage-file-datalake/review/storage-file-datalake.api.md

```diff
@@ -0,0 +1,1188 @@
+  };
+};
+
+// @public (undocumented)
+export interface PathSetMetadataHeaders {
+  // (undocumented)
+  clientRequestId?: string;
+  // (undocumented)
+  date?: Date;
+  // (undocumented)
+  encryptionKeySha256?: string;
+  // (undocumented)
+  etag?: string;
+  // (undocumented)
+  isServerEncrypted?: boolean;
+  // (undocumented)
+  lastModified?: Date;
+  // (undocumented)
+  requestId?: string;
+  // (undocumented)
+  version?: string;
+}
+
+// @public (undocumented)
+export interface PathSetMetadataOptions extends CommonOptions {
+  // (undocumented)
+  abortSignal?: AbortSignalLike_2;
+  // (undocumented)
+  conditions?: DataLakeRequestConditions;
+}
+
+// @public (undocumented)
+export type PathSetMetadataResponse = PathSetMetadataHeaders & {
+  _response: HttpResponse & {
+    parsedHeaders: PathSetMetadataHeaders;
+  };
+};
+
+// @public (undocumented)
+export interface PathSetPermissionsOptions extends CommonOptions {
+  // (undocumented)
+  abortSignal?: AbortSignalLike_2;
+  // (undocumented)
+  conditions?: DataLakeRequestConditions;
+  // (undocumented)
+  group?: string;
+  // (undocumented)
+  owner?: string;
+}
+
+// Warning: (ae-forgotten-export) The symbol "PathUpdateHeaders" needs to be exported by the entry point index.d.ts
+//
+// @public
+type PathUpdateResponse = PathUpdateHeaders & {
+  _response: coreHttp.HttpResponse & {
+    parsedHeaders: PathUpdateHeaders;
+  };
+};
+
+export { PathUpdateResponse as FileAppendResponse };
+
+export { PathUpdateResponse as FileFlushResponse };
+
+// @public
+export class Pipeline {
+  constructor(factories: RequestPolicyFactory[], options?: PipelineOptions);
+  readonly factories: RequestPolicyFactory[];
+  readonly options: PipelineOptions;
+  toServiceClientOptions(): ServiceClientOptions;
+}
+
+// @public
+export interface PipelineOptions {
+  httpClient?: IHttpClient;
+}
+
+// @public (undocumented)
+export type PublicAccessType = "container" | "blob";
+
+export { RequestPolicy };
+
+export { RequestPolicyFactory };
+
+export { RequestPolicyOptions };
+
+export { RestError };
+
+// @public (undocumented)
+export interface RolePermissions {
+  // (undocumented)
+  execute: boolean;
+  // (undocumented)
+  read: boolean;
+  // (undocumented)
+  write: boolean;
+}
+
+// @public (undocumented)
+export interface ServiceGetUserDelegationKeyOptions extends CommonOptions {
+  // (undocumented)
+  abortSignal?: AbortSignalLike;
+}
+
+// @public (undocumented)
+export type ServiceGetUserDelegationKeyResponse = UserDelegationKey &
+  ServiceGetUserDelegationKeyHeaders & {
+    _response: HttpResponse & {
+      parsedHeaders: ServiceGetUserDelegationKeyHeaders;
+      bodyAsText: string;
+      parsedBody: UserDelegationKeyModel;
+    };
+  };
+
+// @public (undocumented)
+export interface ServiceListFileSystemsOptions extends CommonOptions {
+  // (undocumented)
+  abortSignal?: AbortSignalLike;
+  // (undocumented)
+  include?: ListFileSystemsIncludeType;
```

Comment:
Can we change this to a boolean flag `includeMetadata` similar to what we did for blob/file-share etc.?

Reply:
Will do